app/bundles/ApiBundle/Serializer/Driver/ApiMetadataDriver.php

```diff
@@ -40,31 +43,38 @@ class ApiMetadataDriver extends PhpDriver
     /**
      * @var null
      */
-    private $currentPropertyName = null;
+    private $currentPropertyName;
 
     /**
-     * @param \ReflectionClass $class
-     * @param string           $file
+     * @param ReflectionClass $class
      *
-     * @return ClassMetadata
+     * @return \Metadata\ClassMetadata
+     *
+     * @throws ReflectionException
      */
-    protected function loadMetadataFromFile(\ReflectionClass $class, $file)
+    public function loadMetadataForClass(ReflectionClass $class): ?BaseClassMetadata
     {
         if ($class->hasMethod('loadApiMetadata')) {
             $this->metadata = new ClassMetadata($class->getName());
 
-            $this->properties     = [];
-            $this->defaultVersion = '1.0';
-            $this->groupPrefix    = '';
+            $class->getMethod('loadApiMetadata')->invoke(null, $this);
 
-            $serializer = $class->getMethod('loadApiMetadata');
-            $serializer->invoke(null, $this);
+            $metadata = $this->metadata;
 
-            $metadata       = $this->metadata;
-            $this->metadata = null;
+            $this->resetDefaults();
 
             return $metadata;
         }
+
+        return null;
```

Comment:
Nit: I'd like to see an exception thrown here instead of returning null. An exception provides more information about why it couldn't return BaseClassMetadata. But I expect this would require further refactoring and so I'm fine if it stays like this.

Reply:
The base class requires it to not throw an exception and instead return null. The Serializer uses a ChainDriver (where it's a stack of available drivers that it iterates through until they either return a ClassMetadata object or null. If null, it goes on to the next available driver.

https://github.com/schmittjoh/metadata/blob/master/src/Driver/DriverChain.php#L29